src/systemctl/systemctl.c

```diff
@@ -7336,6 +7340,36 @@ static int unit_file_create_new(
         if (r < 0)
                 return r;
 
+        if (original_path) {
+                /* Add a comment with the contents of the original unit */
+                _cleanup_fclose_ FILE *f = NULL;
+                _cleanup_strv_free_ char **lines = NULL;
+                _cleanup_free_ char *commented = NULL;
+
+                f = fopen(original_path, "re");
+                if (!f)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", original_path);
+
+                for (;;) {
+                        _cleanup_free_ char *line = NULL;
+
+                        r = read_line(f, LONG_LINE_MAX, &line);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read \"%s\": %m", original_path);
+                        if (r == 0)
+                                break;
+
+                        r = strv_extend(&lines, line);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read \"%s\": %m", original_path);
```

Comment:
nit: It's fine to directly check return of `strv_extend` and `log_oom()` on error